rust/cymbal/src/symbol_store/caching.rs

```diff
@@ -0,0 +1,121 @@
+use std::{collections::HashMap, sync::Arc, time::Instant};
+
+use axum::async_trait;
+use tokio::sync::Mutex;
```

Comment:
I need to learn what a Mutex is (I'm sure it's something something cross thread)

Reply:
It just means "mutually exclusive" - it's a wrapper around any underlying T, which makes it so you can share access across multiple threads, and know only one mutable reference to the T will ever exist (preventing data races). This specifically is a `tokio::Mutex` because I'm holding the guard across an await point, and holding `std::Mutex` guards ("the lock") across await points causes deadlocks(I can explain in sync why that happens - [tokios docs cover it](https://tokio.rs/tokio/tutorial/shared-state#holding-a-mutexguard-across-an-await) - skip to after the code block)